### DISCUSSION TITLE
segment 크기를 csv 로 내려받을 수 있도록 구현할 때 필요한 것

### DISCUSSION THREAD

**Discussion by acidsound** [Promedius/OHIFViewer#4] [Ideas]
@SaeromB 이슈를 만들고 진행해주시면 됩니다.

먼저 node.js 로 정적 파일을 서빙할 수 있도록 만들고 단일 실행파일로 패키징( https://nodejs.org/api/single-executable-applications.html ) 하는 것을 고려합니다.

/csv 와 같이 route 를 만들고 결과값을 csv 파일을 반환하도록 mock-up 을 작성하세요. mime type을 application/csv로 반환해야겠죠?

OHIFViewer 에 Segmentation 메뉴에서 Export DICOM SEG 아래 Export Segment CSV 와 같은 메뉴를 추가하여 /csv 호출을 통해 받은 결과를 download 받을 수 있도록 하여야 합니다.
![image](https://github.com/Promedius/OHIFViewer/assets/623157/64beb4d2-a700-4a4d-96b0-3136c2bcedc6)

**Comment by SaeromB** [comment]
### download CSV를 했을때 결과값을 받는 과정

1. OHIF Viewer 에 Segmentation 메뉴에서 Export Segment CSV 를 추가한다.
2. Export Segment CSV 선택 시 내부 서버에 Download DICOM SEG 에서 생성한 DICOM 파일을 전달한다.
3. 내부 웹서버에서 DICOM SEG 파일을 dicom2niix (-z n 옵션 사용)를 통해 .nii.gz 로 변환한다.
4. 변환한 .nii.gz 를 MetaMeasurement.exe 에 넣어 .nii.gz.csv 파일을 생성한다.
5. 생성한 .nii.gz.csv 를 내부 웹서버가 반환한다.
6. 내부서버로부터 반환받은 csv 를 OHIF Viewer 가 사용자 local에 csv 파일을 다운로드할 수 있도록 생성한다.
